### PR TITLE
[Shape Inference] Select - shape inference review

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
 - Select shape infer review

### Tickets:
 - 90614

### CODE REVIEW

File: src/core/src/op/select.cpp [hunk 1]
```diff
@@ -41,10 +39,9 @@ void op::v1::Select::validate_and_infer_types() {
                           element::Type::merge(result_et, get_input_element_type(1), get_input_element_type(2)),
                           "Argument 1 and 2 element types must match.");
 
-    std::vector<ov::PartialShape> output_shapes = {ov::PartialShape{}};
-    const std::vector<ov::PartialShape> input_shapes = {get_input_partial_shape(0),
-                                                        get_input_partial_shape(1),
-                                                        get_input_partial_shape(2)};
+    const auto input_shapes = get_node_input_partial_shapes(*this);
+    std::vector<ov::PartialShape> output_shapes{ov::PartialShape::dynamic()};
```
Comment (praasz) [inline src/core/src/op/select.cpp:43]:
```suggestion
    auto output_shapes = std::vector<ov::PartialShape>(1);
```
to use `auto`. Can also add optionally 2nd parameter (`PartialShape{}`) of vector ctor.
It doesn't matter what is inside the vector it must have required number of elements only, `shape_infer` will set correct shape if success. So default initialization is simplest.

**Review by praasz — APPROVED** [review]

File: src/core/tests/type_prop/select.cpp [hunk 2]
```diff
@@ -20,6 +21,154 @@ TEST(type_prop, select_deduce) {
...
+
+    op->validate_and_infer_types();
+
+    EXPECT_EQ(op->get_element_type(), element::f32);
+    EXPECT_EQ(op->get_output_shape(0), (Shape{2, 4}));
+}
+
+TEST(type_prop, select_labels_cond_numpy) {
+    auto labeled_shape = PartialShape{{2, 8}, {3, 7}, {1, 10}, {1, 6}, {1, 10}};
+    set_shape_labels(labeled_shape, {10, 11, 12, 13, 14});
```
[the remark this answers is not in the document]
Comment (mitruska) [reply src/core/tests/type_prop/select.cpp]: Cool feature! Done:)

File: src/plugins/intel_cpu/tests/unit/shape_inference_test/select_shape_inference.cpp [hunk 3]
```diff
@@ -14,35 +14,35 @@ using namespace ov;
 using namespace ov::intel_cpu;
```
[the remark this answers is not in the document]
Comment (mitruska) [reply src/plugins/intel_cpu/tests/unit/shape_inference_test/select_shape_inference.cpp:14]: Done

File: src/core/tests/type_prop/select.cpp [hunk 4]
```diff
@@ -20,6 +21,154 @@ TEST(type_prop, select_deduce) {
...
+
+    const auto& out_shape = op->get_output_partial_shape(0);
+
+    EXPECT_EQ(op->get_element_type(), element::f32);
+    EXPECT_EQ(out_shape, PartialShape({{2, 8}, {3, 7}, -1, 5, -1}));
+    EXPECT_EQ(get_shape_labels(out_shape), expected_labels);
+}
+
+TEST(type_prop, select_labels_all_params_numpy) {
+    auto labeled_shape_cond = PartialShape{-1, 2, 1, 3, 1, {2, 5}, {1, 8}};
```
[the remark this answers is not in the document]
Comment (mitruska) [reply src/core/tests/type_prop/select.cpp]: Added `{-1, 5}` and `{5, -1}`

File: src/core/src/op/select.cpp (hunk 1 above)
Comment (mitruska) [reply src/core/src/op/select.cpp]: Done

**Review by jane-intel — APPROVED** [review]

**Pull request merged**